geth/jail/jail_rpc_test.go

```diff
@@ -147,6 +156,13 @@ func (s *JailRPCTestSuite) TestSendTransaction() {
 	selectErr := s.Account.SelectAccount(TestConfig.Account1.Address, TestConfig.Account1.Password)
 	require.NoError(selectErr)
 
+	// TODO(influx6): How do we move this into ExecutionPolicy.executeSendTransaction since
```

Comment:
I don't really get this comment. Could you elaborate? What would like to move into `ExecutionPolicy.executeSendTransaction`?

Reply:
@adambabik I was considering the fact that we would need to add this piece into the whole approach of sending transactions using the upstream node and from the looks of things, this is not done automatically and must be done for calls when we wish to complete a transaction, like we did here. 

- Is there a way to baked this part as part of the send transaction?
- Are there already a mechanism in place that appopriately calls `CompleteTransaction` with necessary `ChatID` and `password` when running in production?

Reply:
@adambabik Your thoughts?

Reply:
But sending the transaction using the upstream is still two-step process and it works the same way. First, one triggers "send transaction" (it can be done via a C binding, RPC call, `jail.Send` that is bound to `send` function) but it merely puts the transaction into the queue. Next, it must be completed and only then the transaction is actually sent. Again, it works the same way with the upstream. 

Yes, `CompleteTransaction` is provided as a C binding and works in the current develop. With the upstream it should work the same way, just call a proper RPC client. Check out `TxQueueManager.completeRemoteTransaction`. I think you created it in `ExecutionPolicy`, I just copied it.

It can be called manually, however. Check out the following changes:

```
@@ -42,8 +43,9 @@ func TestJailRPCTestSuite(t *testing.T) {

 type JailRPCTestSuite struct {
        BaseTestSuite
-       Account common.AccountManager
-       Policy  *jail.ExecutionPolicy
+       Account        common.AccountManager
+       TxQueueManager common.TxQueueManager
+       Policy         *jail.ExecutionPolicy
 }

@@ -56,6 +58,11 @@ func (s *JailRPCTestSuite) SetupTest() {
        require.NotNil(acctman)

        txQueueManager := node.NewTxQueueManager(nodeman, acctman)
+       txQueueManager.SetTransactionQueueHandler(func(queuedTx *common.QueuedTx) {
+               s.T().Logf("queued a new transaction: %s", queuedTx.ID)
+               _, err := txQueueManager.CompleteTransaction(queuedTx.ID, TestConfig.Account1.Password)
+               s.NoError(err)
+       })

        policy := jail.NewExecutionPolicy(nodeman, acctman, txQueueManager)
        require.NotNil(policy)
@@ -63,6 +70,7 @@ func (s *JailRPCTestSuite) SetupTest() {
        s.Policy = policy
        s.Account = acctman
        s.NodeManager = nodeman
+       s.TxQueueManager = txQueueManager
 }

        // httpRPCServer will serve as an upstream server accepting transactions.
        httpRPCServer := httptest.NewServer(rpcService)
        s.StartTestNode(params.RopstenNetworkID, WithUpstream(httpRPCServer.URL))
        defer s.StopTestNode()

+       s.TxQueueManager.Start()
+       defer s.TxQueueManager.Stop()
```

Reply:
Alright thanks.